cost-events-audit-api.md

```diff
@@ -0,0 +1,113 @@
+
+{% swagger method="get" path="/log" baseUrl="http://<your-kubecost-address>/model/audit/events" summary="Cost events Audit API" %}
+{% swagger-description %}
+Accesses the most recent cluster events and their predicted cost impact
+{% endswagger-description %}
+
+{% swagger-parameter in="path" name="count" type="int" %}
+Number of events to return. If unspecified, it returns all events available.
+{% endswagger-parameter %}
+
+{% swagger-parameter in="path" name="filterNamespaces" type="string" %}
+Comma-separated list of namespaces to match; e.g.
+
+`namespace-one,namespace-two`
+
+will return change events that have occurred only in those two namespaces.
+{% endswagger-parameter %}
+
+{% swagger-parameter in="path" name="filterEventTypes" type="string" %}
+Filter query by event type. Currently, only
+
+`add`
+
+and
+
+`delete`
+
+are accepted. (more types coming soon) Also accepts comma-separated lists, like
+
+`add,delete`
+
+.
+{% endswagger-parameter %}
+
+{% swagger-parameter in="path" name="filterResourceTypes" type="string" %}
+Resource type. Currently, only
+
+`deployment`
+
+ is accepted. (more types coming soon)
+{% endswagger-parameter %}
+
+{% swagger-parameter in="path" name="filterTotalCostLowerBound" type="float" %}
+Floating-point value representing the lower bound for the total event cost.
+{% endswagger-parameter %}
+
+{% swagger-parameter in="path" name="filterTotalCostUpperBound" type="float" %}
+Floating-point value representing the upper bound for the total event cost.
+{% endswagger-parameter %}
+
+{% swagger-response status="200: OK" description="" %}
+```javascript
+{
+    "code": 200,
+    "data": [
+        {
+            "eventType": "add",
+            "resourceType": "deployment",
+            "timestamp": "2023-02-28T13:58:44.552788548Z",
+            "properties": {
+                "cluster": "cluster-one",
+                "name": "sample-deployment",
+                "namespace": "kubecost"
+            },
+            "totalRequests": {
+                "cpu": "100m",
+                "memory": "32Mi"
+            },
+            "resourceCosts": {
+                "costPerCPUCoreHour": 0.031616438356164386,
+                "costPerRAMByteHour": 3.94217363775593e-12,
+                "costPerGPUHour": 0.95
+            },
+            "costPrediction": {
+                "totalMonthlyRate": 2.4045625000000004,
+                "cpuMonthlyRate": 2.3080000000000003,
+                "ramMonthlyRate": 0.0965625,
+                "gpuMonthlyRate": 0
+            }
+        },
+        ...
+    ]
+}
+```
+{% endswagger-response %}
+{% endswagger %}
+
+## Event tracking
+
+Changes at cluster level can range from actions triggered by declarative statements submitted by users (e.g. creation of a Deployment) to automated actions (e.g. cluster autoscaling) or performance events. We detect changes that would have an estimated impact on the overall cluster cost using watchers on the Kubernetes API client.
```

Comment:
> We detect changes that would have an estimated impact on the overall cluster cost using watchers on the Kubernetes API client.

We detect changes that would have an impact on the overall cluster cost using watchers on the Kubernetes API client.